yt/frontends/gadget/data_structures.py

```diff
@@ -561,16 +561,46 @@ def _is_valid(cls, filename, *args, **kwargs):
 
 
 class GadgetHDF5File(ParticleFile):
+
+    _fields_with_cols = ["Metallicity_", "PassiveScalars_", "Chemistry_", "GFM_Metals_"]
+
     def _read_field(self, ptype, field_name, handle=None):
+
+        field_name, col = self._sanitize_field_col(field_name)
+
         if self.total_particles[ptype] == 0:
+            # consider returning an empty array here instead of None
             return
+
         if not handle:
             with self.transaction as handle:
                 v = handle[f"/{ptype}/{field_name}"][self.start : self.end, ...]
         else:
             v = handle[f"/{ptype}/{field_name}"][self.start : self.end, ...]
+
+        if col:
```

Comment:
`if col` will not be True for `col == 0`, so I think this needs some changin'

Reply:
oh right, totally. will fix!